### PR TITLE
[Debugger Expression Evaluation] Enhance function, object method and langlib invocations support using semantic API 

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
This PR improves the existing ballerina debugger expression evaluation engine support for function, method invocation and langlib invocations, by using the new semantic APIs to retrieve function definition information.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/27476.

## Remarks 
However, this implementation is having a blocker as the semantic API supports only a subset of lang libraries (pre-declared) when accessing lang lib function information, resulting `array` and `value` library functions are not supported ATM. Refer https://github.com/ballerina-platform/ballerina-lang/issues/27520 to track the progress.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [x] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [x] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/FunctionInvocationExpressionEvaluator.java [hunk 1]
```diff
@@ -83,69 +87,39 @@ public BExpressionValue evaluate() throws EvaluationException {
...
-                        return Optional.of(new GeneratedStaticMethod(context, cls, method));
-                    }
-                }
-            } catch (ClassNotPreparedException ignored) {
-                // Unprepared classes should be skipped.
-            }
+    private Optional<FunctionSymbol> findFunctionWithinModule() {
+        SemanticModel semanticContext = context.getDebugCompiler().getSemanticInfo();
+        LinePosition position = LinePosition.from(context.getLineNumber(), 0);
+        List<Symbol> functionMatches = semanticContext.visibleSymbols(context.getFileNameWithExt().get(), position)
```
Comment (pubudu91) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/FunctionInvocationExpressionEvaluator.java:93]: If the goal is to find the function from within the constructs in the module itself, you can use the `moduleLevelSymbols()` API.

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/MethodCallExpressionEvaluator.java [hunk 2]
```diff
@@ -91,6 +113,99 @@ public BExpressionValue evaluate() throws EvaluationException {
...
+        Map<String, Value> argValueMap = generateNamedArgs(context, methodName, functionTypeDesc, argEvaluators);
+        langLibMethod.setNamedArgValues(argValueMap);
+        return langLibMethod.invoke();
+    }
+
+    private Optional<ClassSymbol> findClassDefWithinModule(String className) {
+        SemanticModel semanticContext = context.getDebugCompiler().getSemanticInfo();
+        LinePosition position = LinePosition.from(context.getLineNumber(), 0);
+
+        return semanticContext.visibleSymbols(context.getFileNameWithExt().get(), position)
```
Comment (pubudu91) [inline misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/MethodCallExpressionEvaluator.java:195]: Same comment as above here.

**Comment by pubudu91** [comment]
Semantic API usages looks good to me.

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/FunctionInvocationExpressionEvaluator.java (hunk 1 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/FunctionInvocationExpressionEvaluator.java]: +1. Changed in e1c2ab6

File: misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/MethodCallExpressionEvaluator.java (hunk 2 above)
Comment (NipunaRanasinghe) [reply misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/MethodCallExpressionEvaluator.java]: +1. Changed in e1c2ab6

**Review by praveennadarajah — APPROVED** [review]

**Pull request merged**